common/src/gas_provider/mod.rs

```diff
@@ -176,6 +176,29 @@ pub trait Tree {
     /// This can't create imbalance as no value is burned or created.
     fn cut(key: Self::Key, new_key: Self::Key, amount: Self::Balance) -> Result<(), Self::Error>;
 
+    /// Locking some value from underlying balance.
+    ///
+    /// If `key` does not identify any value or the `amount` exceeds what's
+    /// locked under that key, an error is returned.
+    ///
+    /// This can't create imbalance as no value is burned or created.
+    fn lock(key: Self::Key, amount: Self::Balance) -> Result<(), Self::Error>;
+
+    /// Unlocking some value from node's locked balance.
+    ///
+    /// If `key` does not identify any value or the `amount` exceeds what's
+    /// locked under that key, an error is returned.
+    ///
+    /// This can't create imbalance as no value is burned or created.
+    fn unlock(key: Self::Key, amount: Self::Balance) -> Result<(), Self::Error>;
```

Comment:
could you clarify please conditions of unlocking value? You see, currently it's the only function that can block 1) value move to user and 2) tree consumption.

Reply:
discussed